### ISSUE TITLE
DJI MSDK version 4.15 unable to connect to Mavic 2 Enterprise advanced

### ISSUE THREAD

**Issue by fatbringer** [dji-sdk/Mobile-SDK-Android#1106]
I am trying to connect to my Mavic 2 Enterprise advanced using the sample app generated by the MSDK. 
However, the status on the entry screen is:
" No product Connected"
"Product information"
"Firmware version: N/A"

The APP registration has been done successfully. It is able to connect to DJI Spark, and DJI M210 RTK V2.
May I know if there is a problem specific for Mavic 2 Enterprise advance ?

**Comment by dji-dev** [comment]
Agent comment from DJI SDK in <a href='https://djisdksupport.zendesk.com/agent/tickets/60060'>Zendesk ticket #60060</a>:<br><br>尊敬的开发者，感谢您联系DJI 大疆创新
由于github不是我们主要的咨询渠道，您的问题可能跟进不及时。我们建议您通过填写表单（ https://djisdksupport.zendesk.com/hc/zh-cn/requests/new ）向我们反馈问题。或者您也可以在论坛发帖，与其它开发者交流。论坛链接：https://djisdksupport.zendesk.com/hc/zh-cn/community/topics

Dear developer, thank you for contacting DJI.
Since github is not our main consultation channel, your questions may not be followed up in time. We recommend that you fill in the form (https://djisdksupport.zendesk.com/hc/en-us/requests/new) to report problems to us. Or you can post in the forum to communicate with other developers. Forum link: https://djisdksupport.zendesk.com/hc/zh-cn/community/topics
<br><p><em><small>°°°</small></em></p>

**Comment by dji-dev** [comment]
Agent comment from yating.liao in <a href='https://djisdksupport.zendesk.com/agent/tickets/60060'>Zendesk ticket #60060</a>:<br><br>Can you tell me the model of remote control you are using? If it has a screen, it cannot be connected to a smartphone through USB and can only be installed internally.
<br><p><em><small>°°°</small></em></p>

**Comment by fatbringer** [comment]
@dji-dev  i am using the default smart controller that came with the mavic 2 enterprise advance.
Yes it has a screen. 
By installing internally, do you mean to connect the smart controller to the computer with android studio, then installing the app directly onto the controller ?
Is there a need to enable developer options, and other necessary settings ?

**Comment by dji-dev** [comment]
Agent comment from yating.liao in <a href='https://djisdksupport.zendesk.com/agent/tickets/60060'>Zendesk ticket #60060</a>:<br><br>By installing internally, do you mean to connect the smart controller to the computer with android studio, then installing the app directly onto the controller ?
==》Yes.
Is there a need to enable developer options, and other necessary settings ?
==>You need to go into the settings and launch the developer option and turn on usb debugging.
<br><p><em><small>°°°</small></em></p>

**Comment by fatbringer** [comment]
@dji-dev  will we be able to choose which app to run ?
Currently upon start up the controller will execute the pre-installed DJI pilot app when you click on "manual flight".
Will the new app conflict with the pre-installed DJI pilot app ?

**Comment by fatbringer** [comment]
@dji-dev in event of crash, is there a way to reset the controller software ?

**Comment by dji-dev** [comment]
Agent comment from yating.liao in <a href='https://djisdksupport.zendesk.com/agent/tickets/60060'>Zendesk ticket #60060</a>:<br><br>Does the remote control crash? You can press and hold of the power button for more than 20s which will trigger a forced shutdown，then you can try to restart it.
<br><p><em><small>°°°</small></em></p>

**Comment by fatbringer** [comment]
@dji-dev ok thank you. 

Another question is about ActiveTrackMission. For M300 aircraft. Understand that there is no access in MSDK for the "Smart pin and track". Is there access for DJI OSDK on Manifold 2 for that ?

**Comment by dji-dev** [comment]
Agent comment from yating.liao in <a href='https://djisdksupport.zendesk.com/agent/tickets/60060'>Zendesk ticket #60060</a>:<br><br>I asked the relevant colleagues and got the result that OSDK has not an available interface for ActiveTrackMission. You need to develope own ActiveTrackMission yourself.
<br><p><em><small>°°°</small></em></p>

**Comment by dji-dev** [comment]
Agent comment from yating.liao in <a href='https://djisdksupport.zendesk.com/agent/tickets/60060'>Zendesk ticket #60060</a>:<br><br>Sorry,i can't tell you the detail of ActiveTrackMission in account of information security.You can get the video stream from the drone through the MSDK before developing it further.
Which part of the view are you referring to when you say basic view layout? Is it the MSDK SAMPLE?
<br><p><em><small>°°°</small></em></p>

**Comment by fatbringer** [comment]
@dji-dev  yes i am referring to the MSDK sample

for example:
```
public class MoveGimbalWithSpeedView extends BaseThreeBtnView { 
```
and
``` 
public class PushGimbalDataView extends BasePushDataView {
```

**Comment by dji-dev** [comment]
Agent comment from yating.liao in <a href='https://djisdksupport.zendesk.com/agent/tickets/60060'>Zendesk ticket #60060</a>:<br><br>Take MoveGimbalWithSpeedView for example, it inherits from BaseThreeBtnView which is a subclass of LinearLayout, so you can make layout changes in MoveGimbalWithSpeedView through code statements, similar to BaseThreeBtnView.
I hope my answer can help you.
<br><p><em><small>°°°</small></em></p>

**Comment by fatbringer** [comment]
Thank you @dji-dev i will study the code and come back with question.

I have another question about DJI gimbal movement from this article https://developer.dji.com/onboard-sdk/documentation/tutorial/advanced-sensing.html.
It says that: 1 degree/second for 10 pixels. Understand that this is trying to control for gimbal speed.

May I know how did this value come out ? For mapping 10 pixels to 1 degree ?
Will it be different if for example, the image resolution is 1280x720, or 1920x1080 ?
If we want to map gimbal movement instead, i.e. yaw and pitch, can we use the same 10 pixels to 1 degree ?

**Comment by dji-dev** [comment]
Agent comment from kyle.cai in <a href='https://djisdksupport.zendesk.com/agent/tickets/60060'>Zendesk ticket #60060</a>:<br><br>Dear developer ,

Thank you for contacting DJI.

The document you are viewing is an algorithm used by OSDK to track the target, marking the target movement by 10 pixels, and then setting the gimbal to rotate. This value can also be adjusted.

Thank you for your understanding and support, hope you have a nice day.

Best Regards,
DJI SDK Support Team
<br><p><em><small>°°°</small></em></p>

**Comment by fatbringer** [comment]
@dji-dev  yep I understand that it can be adjusted. Is the 10 pixel to 1 degree ratio a recommended value ? And for different image size ?
Because for example, 1920x1080 is much bigger than 1280x720. So maybe the gimbal dont turn enough to the correct position. 
Can you advise on how to find a good value ?

**Comment by dji-dev** [comment]
Agent comment from kyle.cai in <a href='https://djisdksupport.zendesk.com/agent/tickets/60060'>Zendesk ticket #60060</a>:<br><br>Dear developer ,

Thank you for contacting DJI.

Target tracking is to select the target (ROI) and calculate the offset of the ROI to rotate the gimbal, The offset of the ROI is not directly related to the resolution（720p or 1080p）, you can set 5, 10 or other, just start to rotate the gimbal. The gimbal in the OSDK sample is continuously rotating, which is equivalent to 1 deg/s, not 1 deg. Whether the gimbal position is appropriate can be compared by image algorithm.

Thank you for your understanding and support, hope you have a nice day.

Best Regards,
DJI SDK Support Team
<br><p><em><small>°°°</small></em></p>

**Comment by fatbringer** [comment]
@dji-dev ok understand. Thank you !

**Comment by fatbringer** [comment]
@dji-dev  hello. I am unable to deploy my app to the smart controller for mavic 2.
From my android studio,
![Screenshot from 2022-02-22 13-23-22](https://user-images.githubusercontent.com/93758179/155068281-5e55a7c5-3974-4d4a-a21b-fd40c24b30c4.png)

I have already enabled MTP file transfer, and also USB debugging on the smart controller.
I have also made sure to use a data cable for the USB C. 
Could you advise ?

**Comment by kenargo** [comment]
You might be using the incorrect USB configuration, it sounds like you are changing the setting under debug, that’s not the correct setting.  From the app screen, drag down from the top to expose the settings menu, on the far right you will see “USB”, open that and select the top item (Charge).  Once selected, and when connected to the PC you will see Android Studio connect.  No special cable needed.

**Comment by dji-dev** [comment]
Agent comment from yating.liao in <a href='https://djisdksupport.zendesk.com/agent/tickets/60060'>Zendesk ticket #60060</a>:<br><br>This may be related to the phone you are using.
In general, the AS will recognise your device if the usb configuration is set to charging and usb debugging is turned on. If your device is not recognised even in charging mode, you might be able to try RNDIS mode.
<br><p><em><small>°°°</small></em></p>

**Comment by fatbringer** [comment]
@dji-dev i am actually trying to install it to the mavic 2 controller, the one with a built-in screen. I was trying to install it to the controller internally as you had previously mentioned. 

What does AS, and RNDIS mean ? 


@kenargo from what i understand about installing android apps via android studio, the android device must have 1) USB debugging enabled, and 2) MTP protocol (i.e. if you set it to charge only, you cannot install the app) 
I understand what you are suggesting. And I have done what is expected, but android studio is **unable** to detect the mavic 2 controller smart controller with the inbuilt screen. It is able to detect my mobile phone, however, that is not what I want because I want to use the MSDK on the mavic 2 smart controller.

**Comment by dji-dev** [comment]
Agent comment from yating.liao in <a href='https://djisdksupport.zendesk.com/agent/tickets/60060'>Zendesk ticket #60060</a>:<br><br>AS means android studio,and RNDIS is one of the usb configuration.
When the RM500 is connected to the computer via the USB cable, many pop-ups will appear on the RM500. The pop-up asking for Allow USB debugging selects OK, the pop-up prompting for USB configuration (it text message is USB TO) just leave the default as charging. When the selection is done, the Android studio will show that the RM500 has been recognised.
You can refer to my process.
<br><p><em><small>°°°</small></em></p>

**Comment by fatbringer** [comment]
Hi @dji-dev I see this on my smart controller screen. 
![IMG_20220222_131958](https://user-images.githubusercontent.com/93758179/155940274-c89608c9-170c-41d4-9028-19a991eda4e9.jpg)

Is this the correct configuration ?

**Comment by kenargo** [comment]
No. That looks like the developer settings. 

Drag down from the top,  of the home screen,  up will see a little USB icon,  you touch that and choose the top option. 

⁣Get BlueMail for Android ​

On Feb 27, 2022, 11:17 PM, at 11:17 PM, ykn96 ***@***.***> wrote:
>Hi @dji-dev I see this on my smart controller screen. 
>![IMG_20220222_131958](https://user-images.githubusercontent.com/93758179/155940274-c89608c9-170c-41d4-9028-19a991eda4e9.jpg)
>
>Is this the correct configuration ? 
>
>
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/dji-sdk/Mobile-SDK-Android/issues/1106#issuecomment-1053958258
>You are receiving this because you were mentioned.
>
>Message ID:
***@***.***>

**Comment by dji-dev** [comment]
Agent comment from yating.liao in <a href='https://djisdksupport.zendesk.com/agent/tickets/60060'>Zendesk ticket #60060</a>:<br><br>Generally, the RM500 will pop up as shown in the picture when connected to a PC via USB.
If the properties in the status bar are the current properties, I think it should be recognised at this point.
 ![](https://djisdksupport.zendesk.com/attachments/token/jUrNplZttxFEb7SNzjv8g6LTN/?name=image.png)<br><a href='https://djisdksupport.zendesk.com/attachments/token/jUrNplZttxFEb7SNzjv8g6LTN' target='zd_attach'>image.png</a>
<br><p><em><small>°°°</small></em></p>

**Comment by fatbringer** [comment]
@dji-dev hello i am encoutnering this problem instead
![Screenshot from 2022-03-07 12-13-00](https://user-images.githubusercontent.com/93758179/156966839-e0620820-3dc0-4cfd-8e54-c15cd47a169b.png)

Do you know the cause ?

**Comment by dji-dev** [comment]
Agent comment from yating.liao in <a href='https://djisdksupport.zendesk.com/agent/tickets/60060'>Zendesk ticket #60060</a>:<br><br>1.There is a category of cable that can only be used to charge devices. Can the cable you use be used for data transfer?
2.Based on the error message you provided, I have searched for a relevant solution, which is shown below.I hope this can be of some help to you.
 ![](https://djisdksupport.zendesk.com/attachments/token/5OwnBeu4Zg1LZjUbZnlKx4WHT/?name=image.png)<br><a href='https://djisdksupport.zendesk.com/attachments/token/5OwnBeu4Zg1LZjUbZnlKx4WHT' target='zd_attach'>image.png</a>
<br><p><em><small>°°°</small></em></p>

**Comment by fatbringer** [comment]
Hi @dji-dev i am using the data cable. 
For the image, may I know where can I find the setting ?

Also, I realise i can install the app when i run android studio from windows. However, I face the described problem when using android studio on Ubuntu.

**Comment by dji-dev** [comment]
Agent comment from yating.liao in <a href='https://djisdksupport.zendesk.com/agent/tickets/60060'>Zendesk ticket #60060</a>:<br><br>We have not tried to recognize the RM500 using Android studio for Ubuntu. have you tried to recognize other mobile devices in Ubuntu? Android studio may recognize the RM500 in the same way as any other smartphone.
<br><p><em><small>°°°</small></em></p>

**Comment by fatbringer** [comment]
@dji-dev yep i have tried using the android studio with normal android phone. It is able to recognise and install.

**Comment by dji-dev** [comment]
Agent comment from yating.liao in <a href='https://djisdksupport.zendesk.com/agent/tickets/60060'>Zendesk ticket #60060</a>:<br><br>I'm sorry that I can't give you a suitable solution because I haven't tested the recognition on Android studio of ubuntu. Besides, the Android system of RM500 is a custom system of DJI, so it may not be able to adapt to ubuntu.
<br><p><em><small>°°°</small></em></p>

**Comment by fatbringer** [comment]
@dji-dev alright noted. May I know if there are specal things to take note ? since it is a custom system

**Comment by dji-dev** [comment]
Agent comment from yating.liao in <a href='https://djisdksupport.zendesk.com/agent/tickets/60060'>Zendesk ticket #60060</a>:<br><br>I am sorry to have kept you waiting for so long. We have just had a three-day holiday and are only resuming our normal duties today.
In terms of use, there should be nothing special about this remote control, but one thing that will affect development is its inability to connect to a mobile device to control the drone.
<br><p><em><small>°°°</small></em></p>
